third_party/nvfuser/csrc/python_frontend/fusion_record.h

```diff
@@ -1876,6 +1867,63 @@ struct FullOpRecord : RecordFunctor {
   Nvf::DataType dtype_;
 };
 
+struct IotaOpRecord : RecordFunctor {
+  IotaOpRecord(
+      std::vector<State> _args,
+      std::vector<State> _outputs,
+      Nvf::DataType dtype)
+      : RecordFunctor(
+            std::move(_args),
+            std::move(_outputs),
+            "ops.iota",
+            RecordType::IotaOp),
+        dtype_(dtype) {}
+  virtual ~IotaOpRecord() = default;
+  virtual RecordFunctor* clone() final {
+    return new IotaOpRecord(*this);
+  }
+
+  //! Child specific hash function in lower 32 bits.
+  //! | 31 --------------------------------------  0 |
+  //! | Dtype                                        |
+  virtual size_t hash() const final {
+    return RecordFunctor::hash() | static_cast<uint32_t>(dtype_);
+  }
+
+  virtual bool operator==(const RecordFunctor& other) const final {
+    auto result = false;
+    if (auto child_ptr = dynamic_cast<const IotaOpRecord*>(&other)) {
+      result = RecordFunctor::operator==(other) && dtype_ == child_ptr->dtype_;
+    }
+    return result;
+  }
+
+  void operator()(FusionDefinition& fd) final {
+    auto length = fd.getFusionState(args_.at(0).index);
+    auto start = (args_.at(1).stype == StateType::Scalar)
+        ? fd.getFusionState(args_.at(1).index)->as<Nvf::Val>()
+        : nullptr;
+    auto step = (args_.at(1).stype == StateType::Scalar)
```

Comment:
shouldn't this be `2` instead?

Reply:
Please fix this before merging. This also suggests a hole in testing :laughing:

Reply:
You are right. Thanks for catching!